swap_meet/vendor.py

```diff
@@ -1,2 +1,85 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if not item in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
```

Comment:
[suggestion] I like inverting the logic of the condition to be able to treat it like a guard clause, which allows us to outdent the main logic of the function something like the following:

```python
        if my_item not in self.inventory or their_item not in other_vendor.inventory:
            return False

        # Everything else is now less indented
```

Note that this uses [De Morgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). Basically, `x and y` is equal to `not x or not y`